src/components/Result/index.js

```diff
@@ -0,0 +1,37 @@
+import React, {useEffect} from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { ResultItem } from "..";
+import { getRepos } from '../../actions';
+
+export default function Result() {
+    const username = useSelector(state => state.username);
+    const results = useSelector(state => state.results);
+    const loading = useSelector(state => state.loading);
+    const error = useSelector(state => state.error);
+    const dispatch = useDispatch();
+
+    const renderResults = results.map(result => (
+        <ResultItem key={result.id} data={result} />
+    ));
+
+    useEffect(() => {
+        username && dispatch(getRepos(username));
+    }, [username])
```

Comment:
Good use of useEffect but I think this may be why the page gets stuck on loading if the same username is entered twice

Reply:
oh true nice catch, this is exactly why because the loading action gets dispatched but the username doesn't change